src/elements/input/input.svelte

```diff
@@ -95,12 +95,23 @@ const handleInput = () => {
     return;
   }
 
-  if (type === 'number' && input.value.endsWith('.')) {
-    return;
+  if (type === 'number') {
```

Comment:
@micheal-parks would appreciate your eyes on this, not sure if this was the right approach

Reply:
where do we set type? should we not use some typescript validation instead of string checking?

Reply:
this is a parameter to the component called "type" that determines the type of the HTML input element. so, the variable `type` holds a string value of either `"text"`, `"number"`, or `"input"`

Reply:
it depends on the check in [line 128](https://github.com/viamrobotics/prime/pull/204/files#diff-86dbb74c12f51262a7099a9487197a4c34301a0844e630b15d297ca2a0ebbe76R128 )

the current code would fire an event with value `1.`, but wanted to check in with you to see what the desired event firing behavior is. for example, on input string `1.`, should no event be fired or should an event fire with value `1`?

Reply:
I would argue here that we want to set 1. -> 1.0 or 1 the former maybe seems more logical?

Reply:
can I directly compare against the previous value somehow to prevent an eager event fire?

Reply:
@DTCurrie we decided that `1.` will evaluate to `1`, but the event will only be sent if the value is different from the previous.
ex. 
1. `input.value`: `1` --> send value `1`
2. `input.value`: `1.` --> no value sent (repeated)

alternatively,
1. `input.value`: `1.2` --> send value `1.2`
2. `input.value`: `1.` --> send value `1`